test.js

```diff
@@ -49,6 +49,18 @@ describe('os', function() {
   });
 });
 
+describe('osUserInfoNotFound', function() {
+	var os = require('os');
+
+	it('handles ENOENT exceptions', function() {
+		os.userInfo = new Error("ENOENT");
```

Comment:
this looks good, but you'll need to restore `os.userInfo` after this test, or it will pollute future tests.

The most pragmatic approach would be to do the following (I've also fixed up the implementation).


```js
var userInfo = os.userInfo;
try {
  Object.defineProperty(os, 'userInfo', {
    get: function() {
      let error = new Error();
      error.code = 'ENOENT';
      throw error;
     }
  });

  expect(username.os()).to.eql(undefined);
} finally {
  os.userInfo = userInfo;
}
```

Reply:
Actually i'll just quickly fix it up locally :)

Reply:
Thanks! That is really cool. I just learned something new.